### PR TITLE
Mainmenu: Clean up and improve advanced settings dialogues

### PR DESCRIPTION
**Pull request by SmallJoker**
Improvements:
1. Formspec size and description box are calculated last
2. Width and height are now adjustable per setting type
3. Error message (dialogdata.error_message) shortens the description field and is placed below
4. Add more spacing for larger fonts
5. More comments and extensible by setting different height and width values

Settings for screenshot below:
```
font_size = 17
gui_scaling = 0.95
screen_w = 900
screen_h = 600
```

2D-Noise, the free spot on the bottom is for the abs
2D-Noise

Boolean/enum input:
Bool/enum input

Basic field input with wrong format:
wrong format input

Designed to supersede #7791

### CODE REVIEW

**Comment by paramat** [comment]
Looks good, i'll test and review.
Labelled for Android because we can then (in a later PR) be able to tune the size of the int/float formspec to make it more usable on Android, so that Android users are able to set suitable GUI/HUD/font sizes, to attend to #5054.
So replaces #7791

**Comment by paramat** [comment]
Screenshots with game window 1024x600 (default) and all other relevant settings default.

screenshot from 2018-10-20 11-46-42

^ Flags

screenshot from 2018-10-20 11-47-56

^ v3f

screenshot from 2018-10-20 11-51-31

^ path/filepath

It's good that all the wasted space has been removed.
The simpler setting types have formspecs that are too big, but that will be tuned later.

File: builtin/mainmenu/dlg_settings_advanced.lua [hunk 1]
```diff
@@ -674,40 +661,101 @@ local function create_change_setting_formspec(dialogdata)
...
+		comment_text = fgettext_ne(setting.comment)
+	end
+
+	return (
+		"size[" .. width .. "," .. height + 0.25 .. ",true]" ..
+		create_textfield(description_box, setting_name, comment_text) ..
+		formspec ..
+		"button[" .. width / 2 - 2.5 .. "," .. height - 0.25 .. ";2.5,0.75;btn_done;" ..
+			fgettext("Save") .. "]" ..
+		"button[" .. width / 2 .. "," .. height - 0.25 .. ";2.5,0.75;btn_cancel;" ..
```
Comment (paramat) [inline builtin/mainmenu/dlg_settings_advanced.lua]: Button height can't be less than 1.

**Comment by paramat** [comment]
:+1:

**Comment by paramat** [comment]
@SmallJoker perhaps we can now detect Android and reduce the size of the int/float formspec (and perhaps bool and enum too) for that platform?

**Comment by SmallJoker** [comment]
@paramat Rather check why the GUI is scaled so badly. The font usually displayed right on Android, but the formspec size does not seem to be adjusted well to the maximal height and width of the screen.

**Comment by paramat** [comment]
The main menu pages scale with game window size, but each setting value edit formspec doesn't, certainly this needs investigation.

**Pull request merged**